src/envoy/http/authn/authenticator_base_test.cc

```diff
@@ -60,7 +62,67 @@ TEST_F(AuthenticatorBaseTest, ValidateX509) {
                               });
 }
 
-// TODO: more tests for other cases of x509 and Jwt.
+TEST_F(AuthenticatorBaseTest, ValidateX509OnSslConnectionWithNoPeerCert) {
+  iaapi::MutualTls mTlsParams;
+  EXPECT_CALL(Const(connection_), ssl()).Times(2).WillRepeatedly(Return(&ssl_));
```

Comment:
why this is Times(2) and many other places are Times(3). 
is it important to verify 2, or 3?  Is it implementation detail.   Usually we don't test implementation detail.  This make test flaky.

Reply:
Not too much for now. Remove.